container/lxd/server.go

```diff
@@ -299,5 +302,21 @@ func (s *Server) HostArch() string {
 // IsLXDNotFound checks if an error from the LXD API indicates that a requested
 // entity was not found.
 func IsLXDNotFound(err error) bool {
-	return err != nil && err.Error() == "not found"
+	if err == nil {
+		return false
+	}
+
+	// TODO (stickupkid): This is a hack until we can correctly upstream the
+	// lxd project.
+	if statusErr, ok := errors.Cause(err).(lxdStatusError); ok && statusErr.Status() == http.StatusNotFound {
+		return true
+	}
+
+	return strings.ToLower(err.Error()) == "not found"
```

Comment:
Maybe `strings.Contains(strings.ToLower(err.Error()), "not found")` would be a bit safer against future changes.